airflow/executors/base_executor.py

```diff
@@ -55,7 +55,7 @@ class BaseExecutor(LoggingMixin):
         ``0`` for infinity
     """
 
-    job_id: Optional[str] = None
+    job_id: Union[None, int, str] = None
```

Comment:
Maybe:

```suggestion
    job_id: Optional[Union[int, str]] = None
```

A bit longer, but I think conceptually more consistent with other optionals.